core/modules/qproc/ChunkSpec.cc

```diff
@@ -57,12 +60,116 @@ std::ostream& operator<<(std::ostream& os, ChunkSpec const& c) {
     return os;
 }
 
+
+/// ChunkSpecVector intersection.
+/// Computes ChunkSpecVector intersection, overwriting dest with the result.
+///
+/// precondition: Elements in ChunkSpecVector should be minimized in the sense
+/// that there should be only one element for a particular chunkId. i.e., for
+/// all ChunkSpec A element, there is no ChunkSpec B element in the same vector
+/// where A.chunkId == B.chunkId.
+void intersectSorted(ChunkSpecVector& dest, ChunkSpecVector const& a) {
+    ChunkSpecVector tmp;
+    tmp.swap(dest);
```

Comment:
I think this would be clearer if you did the swap with dest right before returning. More importantly, that strategy yields the strong exception safety guarantee.